crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -58,14 +59,17 @@ impl<'me, 'ast> Visit<'ast> for AstScanner<'me> {
           }
         };
         match object_symbol_in_top_level {
-          // import statements are hoisted to the top of the module, so in this time being, all imports scanned.
-          Some(sym_ref) if self.result.named_imports.contains_key(&sym_ref) => {
+          // - Import statements are hoisted to the top of the module, so in this time being, all imports are scanned.
+          // - Having empty span will also results to bailout since we rely on span to identify ast nodes.
+          Some(sym_ref)
+            if self.result.named_imports.contains_key(&sym_ref) && !expr.span().is_unspanned() =>
```

Comment:
What if other plugin inserted a top-level `MemberExpr` in **transform** phase?

Reply:
> Import statements are hoisted to the top of the module, so in this time being, all imports are scanned.

No matter what transform, this rule still applies.

> except that using span as a key might be a footgun in the future.

As for invalid span, we just do bailout. We'll switch `AstNoeId` once oxc stablize it.

Reply:
`AstNodeId` sounds more robust

Reply:
I saw a litlte issue when I try to inserted a ZERO SPAN member expr, e.g.
```js
// index.js
import * as test from './test.js'
console.log(test.a)

test.b() // `test.b()` inserted by transformer

// test.js
export const a = 1000;
export const b = 10000;

```
It would ref the whole namespace? is that expected? 
**output** 
```js

//#region rolldown:runtime
var __defProp = Object.defineProperty;
var __export = (target, all) => {
	for (var name in all) __defProp(target, name, {
		get: all[name],
		enumerable: true
	});
};

//#endregion
//#region hyper-cube.js
var hyper_cube_ns = {};
__export(hyper_cube_ns, {
	a: () => a,
	b: () => b
});
const a = 100;
const b = 1000;
test.b();

//#endregion
//#region entry.js
a();
hyper_cube_ns.b();

//#endregion
//# sourceMappingURL=entry.js.map
```

Reply:
I saw a litlte issue when I try to inserted a ZERO SPAN member expr, e.g. 
```js
// MULTIPLE ENTRY MODULES
import * as test from './hyper-cube.js';

test.a()
test.b() // inserted by transformer.
// hyper-cube.js
export const a = 100;
export const b = 1000;

```
It would ref the whole namespace.
**output** 
```js

//#region rolldown:runtime
var __defProp = Object.defineProperty;
var __export = (target, all) => {
	for (var name in all) __defProp(target, name, {
		get: all[name],
		enumerable: true
	});
};

//#endregion
//#region hyper-cube.js
var hyper_cube_ns = {};
__export(hyper_cube_ns, {
	a: () => a,
	b: () => b
});
const a = 100;
const b = 1000;

//#endregion
//#region entry.js
a();
hyper_cube_ns.b();

//#endregion
//# sourceMappingURL=entry.js.map
```

Reply:
the result is still right, just wait for `AstNodeId` stablized.

Reply:
Yeah. It's expected. We have code to prevent scanning such member expr by `!expr.span().is_unspanned()`.